src/components/member-page/donation/table.js

```diff
@@ -0,0 +1,203 @@
+// components
+import { StatusBadge } from './status-bagde'
+
+const TableContainer = styled.div`
+  width: 100%;
+  ${mq.mobileOnly`
+    padding: 24px 0px 64px;
+  `}
+  ${mq.tabletAndAbove`
+    display: grid;
+    grid-template-columns: repeat(6, 1fr);
+    grid-column-gap: 8px;
+    padding: 32px 0px 120px;
+  `}
+`
+
+const TableHeader = styled.div`
+  display: contents;
+  .donation-date {
+    grid-column: 1 / 2;
+  }
+  .subject {
+    grid-column: 2 / 3;
+  }
+  .donation-number {
+    grid-column: 3 / 5;
+  }
+  .amount {
+    grid-column: 5 / 6;
+  }
+  .status {
+    grid-column: 6 / 7;
+  }
+  ${mq.mobileOnly`
+    display: none;
+  `}
+`
+
+const DividerContainer = styled.div`
+  width: 100%;
+  grid-column: 1 / 7;
+  margin-top: 16px;
+  margin-bottom: 16px;
+`
+
+const TableContent = styled.div`
+  display: contents;
+  /* phase 2 */
+  /* cursor: pointer; */
+  .donation-date {
+    grid-column: 1 / 2;
+  }
+  .type {
+    grid-column: 2 / 3;
+  }
+  .donation-number {
+    grid-column: 3 / 5;
+    width: 100%;
+    display: block;
+    word-break: break-all;
+  }
+  .amount {
+    grid-column: 5 / 6;
+  }
+  .status {
+    grid-column: 6 / 7;
+    display: flex;
+    align-items: center;
+  }
+  ${mq.mobileOnly`
+    display: none;
+  `}
+`
+
+const MobileTableContent = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 8px;
+  align-self: stretch;
+  .row {
+    width: 100%;
+    display: flex;
+    gap: 8px;
+    .amount {
+      display: flex;
+      flex-grow: 1;
+      justify-content: end;
+    }
+  }
+  .donation-number {
+    width: 100%;
+    display: block;
+    word-break: break-all;
+  }
+  ${mq.tabletAndAbove`
+    display: none;
+  `}
+`
+
+const P1Gray800 = styled(P1)`
+  color: ${colorGrayscale.gray800};
+`
+
+const P1Gray600 = styled(P1)`
+  color: ${colorGrayscale.gray600};
+`
+
+const formattedDate = date =>
+  `${date.getUTCFullYear()}/${date.getUTCMonth() + 1}/${date.getUTCDate()}`
+
+const TableRow = memo(({ record }) => {
+  const {
+    type,
+    created_at: createdAt,
+    amount,
+    status,
+    order_number: orderNumber,
+  } = record
+  const donationDate = formattedDate(new Date(createdAt))
+  const amountSuffix = type === DonationType.PRIME ? '元' : '元/月'
+
+  return (
+    <React.Fragment>
+      <TableContent>
+        <P1Gray800 className="donation-date" text={donationDate} />
+        <P1Gray800 className="type" text={DonationTypeLabel[type]} />
+        <P1Gray800 className="donation-number" text={orderNumber} />
+        <P1Gray800
+          className="amount"
+          text={`${amount.toLocaleString('en-US')}${amountSuffix}`}
+        />
+        <div className="status">
+          <StatusBadge status={status} type={type} />
+        </div>
+      </TableContent>
+      <MobileTableContent>
+        <P1Gray800 text={donationDate} />
+        <div className="row">
```

Comment:
tip: styled components could reference each other ([ref](https://styled-components.com/docs/advanced#referring-to-other-components))

Reply:
❌